app/src/main/java/de/dreier/mytargets/base/db/migrations/Migration26.kt

```diff
@@ -28,55 +28,59 @@ object Migration26 : Migration(25, 26) {
+                .useEach { round ->
+                    val diameter = Dimension.parse(round.getString(3))
+                    val target = Target(round.getLong(1), round.getInt(2), diameter)
+                    val roundId = round.getLong(0)
 
-                database.query("SELECT `id` FROM `End` WHERE `roundId` = $roundId")
-                    .useEach { end ->
-                        val endId = end.getLong(0)
+                    database.query("SELECT `id` FROM `End` WHERE `roundId` = $roundId")
+                        .useEach { end ->
+                            val endId = end.getLong(0)
 
-                        val shots = mutableListOf<Shot>()
-                        database.query("SELECT `index`, `scoringRing` FROM `Shot` WHERE `endId` = $endId ORDER BY `index`")
-                            .useEach { shotCursor ->
-                                shots.add(
-                                    Shot(
-                                        index = shotCursor.getInt(0),
-                                        scoringRing = shotCursor.getInt(1)
+                            val shots = mutableListOf<Shot>()
+                            database.query("SELECT `index`, `scoringRing` FROM `Shot` WHERE `endId` = $endId ORDER BY `index`")
+                                .useEach { shotCursor ->
+                                    shots.add(
+                                        Shot(
+                                            index = shotCursor.getInt(0),
+                                            scoringRing = shotCursor.getInt(1)
+                                        )
                                     )
-                                )
-                            }
-
-                        val score = target.getReachedScore(shots)
-                        database.execSQL(
-                            "UPDATE `End` SET " +
-                                    "`reachedPoints` = ${score.reachedPoints}, " +
-                                    "`totalPoints` = ${score.totalPoints}, " +
-                                    "`shotCount` = ${score.shotCount} " +
-                                    "WHERE `id` = $endId"
-                        )
-                    }
-            }
+                                }
+
+                            val score = target.getReachedScore(shots)
+                            database.execSQL(
+                                "UPDATE `End` SET " +
+                                        "`reachedPoints` = ${score.reachedPoints}, " +
+                                        "`totalPoints` = ${score.totalPoints}, " +
+                                        "`shotCount` = ${score.shotCount} " +
+                                        "WHERE `id` = $endId"
+                            )
+                        }
+                }
+        } catch (e: Exception) {
+            Timber.i(e, "DB has changed because of restoring a backup")
```

Comment:
Catching generic exceptions without properly handling them is basically always a bad idea. I also don't understand yet how this would solve the restore problems.

Basically when you import a backup into the app the database contained in the backup is stored in the internal data of the app. Then the app restarts. If a database is detected that is currently being imported the previous database is replaced by the temporary copy (imported db). When the imported database is on an older version schema it has to be migrated to the newest schema (which happens automatically). Skipping the migration (as you did here) in case an error occurs just fixes the symptom of the app crashing, but not the underlying problem.

Reply:
The problem is that the version of backup DB is 25 and it tries to migrate to 26. but, Training table has already all columns!